lib/nfg_onboarder/onboarding_controller_helper.rb

```diff
@@ -20,18 +20,22 @@ module OnboardingControllerHelper
       expose(:last_step) { steps.last == step }
       expose(:locale_namespace) { self.class.name.underscore.gsub('_controller', '').split('/') }
       expose(:form_params) { params.fetch("#{field_prefix}_#{step}", {}).permit! }
+      expose(:exit_without_saving?) { exit_without_save_steps.include?(onboarding_session.current_step) }
+      expose(:exit_with_saving?) { exit_with_save_steps.include?(onboarding_session.current_step) }
 
       def show
         on_before_show
         render_wizard
       end
 
       def update
+        redirect_to finish_wizard_path and return if exit_without_saving? && exit?
         if form.validate(form_params)
           on_before_save
           form.save
           on_valid_step
           process_on_last_step if last_step
+          redirect_to finish_wizard_path and return if exit? && (exit_with_saving? || (!exit_without_saving? && !exit_with_saving?))
```

Comment:
@hoenth  I updated this logic here.  if hte step is neither in exit withotu saving nor in exit_with saving, then it should save and exit. updated generator comment as well.  this is just in case someone forgets to add a step to these methods.

Reply:
@pathakh Can you write the logic out in english as a comment. My brain is having a hard time comprehending all of the negatives and double negatives. what specifically is `(!exit_without_saving? && !exit_with_saving?)` checking for. if you were looking at this code 6 months from now, would you know immediately what you meant this to do?

Reply:
@hoenth I see what you mean by that.  I simplified it more.  I don't think we need to track steps that should save when exiting.  we should only track steps that shouldn't save when exiting, everything else should just save and exit.  This reduced the logic to a simple statement.